python/cudf/cudf/core/column/string.py

```diff
@@ -3752,8 +3752,9 @@ def endswith(self, pat: str) -> SeriesOrIndex:
         dtype: bool
         """
         if pat is None:
-            result_col = column.column_empty(
-                len(self._column), dtype="bool", masked=True
+            raise TypeError(
+                f"expected a string or a sequence-like object, not "
```

Comment:
Note, in 1.5. we also further restricted `startswith`/`endswith` to only accept a tuple specifically to mirror the stdlib

```
In [15]: "f".endswith(["f", "g"])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [15], in <cell line: 1>()
----> 1 "f".endswith(["f", "g"])

TypeError: endswith first arg must be str or a tuple of str, not list
```

Reply:
I did see that change, I'll be opening a follow-up PR because that will need some discussions probably.